pkg/resource/deploy/step_generator.go

```diff
@@ -1949,6 +1949,11 @@ func (sg *stepGenerator) GenerateDeletes(targetsOpt UrnTargets, excludesOpt UrnT
 				continue
 			}
 
+			if forbiddenResourcesToDelete != nil && forbiddenResourcesToDelete[res.URN] {
```

Comment:
Isn't this just the same as what isTargeted is checking? And now your not adding a same step for it, like we should do at line ~2012